pkg/controller/bootstrap/bootstrapsigner.go

```diff
@@ -0,0 +1,277 @@
+			},
+		},
+		&v1.ConfigMap{},
+		options.ConfigMapResync,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    func(_ interface{}) { e.signConfigMap() },
+			UpdateFunc: func(_, _ interface{}) { e.signConfigMap() },
+		},
+	)
+
+	secretSelector := fields.SelectorFromSet(map[string]string{api.SecretTypeField: string(api.SecretTypeBootstrapToken)})
+	e.secrets, e.secretsController = cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo v1.ListOptions) (runtime.Object, error) {
+				lo.FieldSelector = secretSelector.String()
+				return e.client.Core().Secrets(api.NamespaceSystem).List(lo)
+			},
+			WatchFunc: func(lo v1.ListOptions) (watch.Interface, error) {
+				lo.FieldSelector = secretSelector.String()
+				return e.client.Core().Secrets(api.NamespaceSystem).Watch(lo)
+			},
+		},
+		&v1.Secret{},
+		options.SecretResync,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    func(_ interface{}) { e.signConfigMap() },
+			UpdateFunc: func(_, _ interface{}) { e.signConfigMap() },
+			DeleteFunc: func(_ interface{}) { e.signConfigMap() },
+		},
+	)
+	return e
+}
+
+// Run runs controller loops and returns immediately
+func (e *BootstrapSigner) Run(stopCh <-chan struct{}) {
+	go e.configMapsController.Run(stopCh)
+	go e.secretsController.Run(stopCh)
+	<-stopCh
+}
+
+// signConfigMap computes the signatures on our latest cached objects and writes
+// back if necessary.
+func (e *BootstrapSigner) signConfigMap() {
+	origCM := e.getConfigMap()
+
+	if origCM != nil {
+		var needUpdate = false
+
+		newCM, err := copyConfigMap(origCM)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		}
+
+		// First capture the config we are signing
+		content, ok := newCM.Data[kubeConfigKey]
+		if !ok {
+			glog.V(3).Infof("No %s key in %s/%s ConfigMap", kubeConfigKey, origCM.Namespace, origCM.Name)
+			return
+		}
+
+		// Next remove and save all existing signatures
+		sigs := map[string]string{}
+		for key, value := range newCM.Data {
+			if strings.HasPrefix(key, signaturePrefix) {
+				tokenID := strings.TrimPrefix(key, signaturePrefix)
+				sigs[tokenID] = value
+				delete(newCM.Data, key)
+			}
+		}
+
+		// Now recompute signatures and store them on the new map
+		tokens := e.getTokens()
+		for tokenID, tokenValue := range tokens {
+			sig, err := computeDetachedSig(content, tokenID, tokenValue)
+			if err != nil {
+				utilruntime.HandleError(err)
+			}
+
+			// Check to see if this signature is changed or new.
+			oldSig, _ := sigs[tokenID]
+			if sig != oldSig {
+				needUpdate = true
+			}
+			delete(sigs, tokenID)
+
+			newCM.Data[signaturePrefix+tokenID] = sig
+		}
+
+		// If we have signatures left over we know that some signatures were
+		// removed.  We now need to update the ConfigMap
+		if len(sigs) != 0 {
+			needUpdate = true
+		}
+
+		if needUpdate {
+			e.updateConfigMap(newCM)
+		}
+	}
+}
+
+func (e *BootstrapSigner) updateConfigMap(cm *v1.ConfigMap) {
+	for i := 0; i < e.maxRetries; i++ {
```

Comment:
Worst cas this is going to cause 10 very rapid retries to the apiserver. I suggest you backoff here. https://godoc.org/k8s.io/kubernetes/pkg/util/wait#ExponentialBackoff